### PR TITLE
Ignore display name for policy_ldap_identity_source when NSX version >= 4.2.1

### PR DESCRIPTION
**Pull request by 2ez4szliu**
Fix an issue with NSX version >= 4.2.1 that ldap_identity_source without display_name cannot be successfully created because NSX has forced the display_name to match its ID.

### CODE REVIEW

File: nsxt/resource_nsxt_policy_ldap_identity_source.go [hunk 1]
```diff
@@ -219,6 +220,9 @@ func resourceNsxtPolicyLdapIdentitySourceProbeAndUpdate(d *schema.ResourceData,
 			LdapServers:            ldapServers,
 			ResourceType:           nsxModel.LdapIdentitySource_RESOURCE_TYPE_ACTIVEDIRECTORYIDENTITYSOURCE,
 		}
+		if util.NsxVersionLower("4.2.1") {
```
Comment (annakhm) [inline nsxt/resource_nsxt_policy_ldap_identity_source.go:223]: This does not seem to change behavior because DisplayName is still assigned in line 212

**Comment by annakhm** [comment]
Hi @2ez4szliu, can you please explain in more detail what NSX behavior we're trying to work around here?

File: nsxt/resource_nsxt_policy_ldap_identity_source.go (hunk 1 above)
Comment (2ez4szliu) [reply nsxt/resource_nsxt_policy_ldap_identity_source.go:222]: Thanks, it's fixed

**Comment by 2ez4szliu** [comment]
> Hi @2ez4szliu, can you please explain in more detail what NSX behavior we're trying to work around here?

Hi @annakhm, the issue is 3403819, if I understand correctly, before 421 we can specify arbitrary display_name for ldap_identity_source, but since 421 it has to match a specific value. As @salv-orlando suggested we can just ignore parsing display name on provider side, in that case the NSX resource created will use the correct name by default

**Comment by ksamoray** [comment]
As 4.2.1 isn't released yet, can't it be resolved within the platform?

**Comment by annakhm** [comment]
I think its preferrable to remove display name from the resource (it is still in Beta)

**Comment by annakhm** [comment]
Suggested alternative https://github.com/vmware/terraform-provider-nsxt/pull/1254

**Comment by salv-orlando** [comment]
Closing this PR as it is superseded by PR #1254.
Thanks a lot @2ez4szliu for your contribution!

**Pull request closed without merging**